packages/sil-governance/src/routes/dashboard/components/Proposal/VotingPanel/Buttons.tsx

```diff
@@ -48,13 +47,6 @@ const Buttons = ({ id, vote }: Props): JSX.Element => {
         toast.show(communicationTexts.notReadyMessage, ToastVariant.ERROR);
       } else {
         dispatch(setTransactionsStateAction(transactionId));
-
-        setTimeout(() => {
-          getProposals(governor).then(
-            chainProposals => dispatch(replaceProposalsAction(chainProposals)),
-            error => console.error(error),
-          );
-        }, 5000);
```

Comment:
this is a good place to set the `needsUpdate` flag (described below) to true as we are expecing changes

Reply:
Should it be set with the timeout of 5 seconds or inmediately after `dispatch(setTransactionsStateAction(transactionId));`?

Reply:
Very good question. We expect a change on the blockchain shortly after running this code. What about setting it twice, immediately and again after 5 seconds